### PR TITLE
Fix MIME type errors from the AuthController

### PR DESCRIPTION
**Pull request by joshbaskaran**
Enhanced the GET mappings for "/token" and "/user" endpoints in the AuthController. Now these endpoints consume "application/json" requests and specify their respective produced data formats. This provides a more defined behavior when these endpoints are accessed.

Potential fix for MIME type errors being thrown.

This should be merged regardless of whether it fixes the MIME errors or not. It still makes the code more robust.

In the event that we are still facing error being thrown, we might also catch the Exception and return a more appropriate error message to the caller.

### CODE REVIEW

**Comment by kjellp** [comment]
Reviewing the code by eye I think it looks good :) How much have you been able to test this new feature as part of the test suite? (I wasnt really able to see from the logs if any of the tests are testing this). Let me know if you think it needs to be tested on the tryggve server  before a merge, or this second pair of eyes looking at the code and tests completing green should suffice.

**Comment by joshbaskaran** [comment]
This is unfortunately hard to test. Since we don't know what request triggered the MIME error to begin with. The best we can do is keep an eye on prod to see if we catch that error again (should it persist) then we simply create a controller advice so that in the event of such an exception occurring again, we catch the exception, and return a 400 bad request status instead.

**Review by kjellp — APPROVED** [review]

**Pull request merged**
